lib/device.js

```diff
@@ -1082,7 +1155,7 @@ export default class Device {
 
       // TODO: please consider array uniform: uniformInfo.size > 0
 
-      let commitFunc = _type2uniformCommit[uniformInfo.type];
+      let commitFunc = (uniformInfo.size === undefined) ? _type2uniformCommit[uniformInfo.type] : _type2uniformArrayCommit[uniformInfo.type];
```

Comment:
I think `commitFunc(gl, uniformInfo.location, uniform.value, uniformInfo.size !== undefined)` is better